haystack/pipelines/base.py

```diff
@@ -18,14 +21,51 @@
     ray = None  # type: ignore
     serve = None  # type: ignore
 
-from haystack.schema import MultiLabel, Document
+from haystack.schema import Label, MultiLabel, Document
 from haystack.nodes.base import BaseComponent
 from haystack.document_stores.base import BaseDocumentStore
 
 
 logger = logging.getLogger(__name__)
 
 
+class EvaluationResult:
+    def __init__(self, node_results: Dict[str, DataFrame] = None) -> None:
+        self.node_results: Dict[str, DataFrame] = {} if node_results is None else node_results
+
+    def __getitem__(self, key: str):
+        return self.node_results.__getitem__(key)
+
+    def __delitem__(self, key: str):
+        self.node_results.__delitem__(key)
+
+    def __setitem__(self, key: str, value: DataFrame):
+        self.node_results.__setitem__(key, value)
+
+    def __contains__(self, key: str):
+        return self.node_results.keys().__contains__(key)
+
+    def append(self, key: str, value: DataFrame):
+        if key in self.node_results:
+            self.node_results[key] = pd.concat([self.node_results[key], value])
+        else:    
+            self.node_results[key] = value
+
+    def save(self, out_dir: Union[str, Path]):
+        out_dir = out_dir if isinstance(out_dir, Path) else Path(out_dir)
+        for node_name, df in self.node_results.items():
+            target_path = out_dir / f"{node_name}.csv"
+            df.to_csv(target_path, index=False, header=True)
+
+    @classmethod
+    def load(cls, load_dir: Union[str, Path]):
+        load_dir =  load_dir if isinstance(load_dir, Path) else Path(load_dir)
+        csv_files = [file for file in load_dir.iterdir() if file.is_file() and file.suffix == ".csv"]
+        node_results = {file.name[:-len(file.suffix)]: pd.read_csv(file, header=0) for file in csv_files}
```

Comment:
`file.stem` can be used instead of `file.name[:-len(file.suffix)]`

Reply:
done